### PR TITLE
Fix spacing for arrow functions

### PR DESCRIPTION
**Pull request by krzysztofrewak**
Now arrow functions spacing is broken and it's working like this:
```php
array_filter([1, 2, 3, 4], fn (int $i): bool => $i % 2 === 0);
```

PHPStorm and other web sources have something like this:
```php
array_filter([1, 2, 3, 4], fn(int $i): bool => $i % 2 === 0);
```

I took care about previous PHP versions (without `T_FN` token) too.

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
This is not a "fix" but a breaking change, as we can see from the tests.

**Comment by GrahamCampbell** [comment]
We have explict tests that expect the space to be added in there.

**Comment by krzysztofrewak** [comment]
> There are explicit tests that expect the space to be added in there.

@GrahamCampbell, do you want me to change these tests? Or are you questioning the legitimacy of these changes? In PHP docs there're multiple cases with `fn()` withou spaces: https://www.php.net/manual/en/functions.arrow.php:

```php

<?php

$y = 1;
 
$fn1 = fn($x) => $x + $y;
// equivalent to using $y by value:
$fn2 = function ($x) use ($y) {
    return $x + $y;
};

var_export($fn1(3));
?>
```

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]
👎 This is a breaking change. At best, this could be changed to be a config option, but certainly not the default changed.

File: tests/Fixtures/Integration/misc/PHP8_0.test [hunk 1]
```diff
@@ -112,7 +112,7 @@ class T_Mixed
 if (1) {
     foo() ?? throw new \Exception();
     $a = $condition && throw new Exception();
-    $callable = fn () => throw new Exception();
```
Comment (GrahamCampbell) [inline tests/Fixtures/Integration/misc/PHP8_0.test:115]: Breaking change

File: tests/Fixtures/Integration/misc/PHP7_4.test [hunk 2]
```diff
@@ -46,7 +46,7 @@ class Foo {
     var ?Foo\Bar $qux;
 }
 // https://wiki.php.net/rfc/arrow_functions_v2
-$foo = fn ($v) => (string) $v;
```
Comment (GrahamCampbell) [inline tests/Fixtures/Integration/misc/PHP7_4.test:49]: Breaking change.

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

File: src/Fixer/FunctionNotation/NoSpacesAfterFunctionNameFixer.php [hunk 3]
```diff
@@ -141,7 +141,7 @@ private function getBraceAfterVariableKinds()
      *
      * @return int[] Token names
      */
-    private function getFunctionyTokenKinds()
+    private function getFunctionTokenKinds()
```
Comment (kubawerlos) [inline src/Fixer/FunctionNotation/NoSpacesAfterFunctionNameFixer.php:144]: Why did you rename the function? The new name is misleading.

**Review by kubawerlos — CHANGES_REQUESTED** [review]
The tests you have changed (and the code you did not, which makes the build failing) are there exactly to check the space is added.

It was done in [this](https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/4600) PR, see especially file `tests/Fixer/FunctionNotation/FunctionDeclarationFixerTest.php`.

Even if we want this behaviour to be allowed, it should not be in `NoSpacesAfterFunctionNameFixer` as the arrow function does not have name, but in `FunctionDeclarationFixer` - most likely as option `none`/`single` (by default `single`) added.

**Pull request closed without merging**